cpp/src/arrow/compute/kernels/scalar_boolean.cc

```diff
@@ -175,15 +175,24 @@ struct KleeneAnd : Commutative<KleeneAnd> {
     }
 
     if (right_true) {
-      GetBitmap(*out, 0).CopyFrom(GetBitmap(left, 0));
+      if (left.GetNullCount() == 0) {
+        GetBitmap(*out, 0).SetBitsTo(true);
```

Comment:
How about instead:
```c++
out->null_count = 0;
out->buffers[0] = nullptr;
```

Reply:
Done.
Looks it's better to tell the scalar kernel framework not to pre-allocate the valid buffer.